crates/goose/src/agents/reference.rs

```diff
@@ -65,7 +68,60 @@ impl Agent for ReferenceAgent {
+        let read_resource_tool = Tool::new(
+            "platform__read_resource".to_string(),
+            indoc! {r#"
+                Read a resource from a system.
+
+                Resources allow systems to share data that provide context to LLMs, such as
+                files, database schemas, or application-specific information. This tool searches for the
+                resource URI in the provided system, and reads in the resource content. If no system
+                is provided, the tool will search all systems for the resource.
+
+                The read_resource tool is typically used with a search query (can be before or after). Here are two examples:
+                1. Search for files in Google Drive MCP Server, then call read_resource(gdrive:///<file_id>) to read the Google Drive file.
+                2. You need to gather schema information about a Postgres table before creating a query. So you call read_resource(postgres://<host>/<table>/schema)
+                    to get the schema information for a table and then use to construct your query.
+            "#}.to_string(),
+            json!({
+                "type": "object",
+                "required": ["uri"],
+                "properties": {
+                    "uri": {"type": "string", "description": "Resource URI"},
+                    "system_name": {"type": "string", "description": "Optional system name"}
+                }
+            }),
+        );
+
+        let list_resources_tool = Tool::new(
+            "platform__list_resources".to_string(),
+            indoc! {r#"
+                List resources from a system(s).
+
+                Resources allow systems to share data that provide context to LLMs, such as
+                files, database schemas, or application-specific information. This tool lists resources
+                in the provided system, and returns a list for the user to browse. If no system
+                is provided, the tool will search all systems for the resource.
+
+                The list_resources tool is typically used before a read_resource tool call. Here are two examples:
+                1. List files in Google Drive MCP Server, then call read_resource(gdrive:///<file_id>) to read the Google Drive file.
+                2. You want to see what tables exist in Postgre so you can find schema information about that table before creating a query. So you call list_resources to see whats available then you call read_resource(postgres://<host>/<table>/schema)
+                    to get the schema information for a table and then use to construct your query.
+            "#}.to_string(),
+            json!({
+                "type": "object",
+                "properties": {
+                    "system_name": {"type": "string", "description": "Optional system name"}
+                }
+            }),
+        );
+
+        tools.push(read_resource_tool);
```

Comment:
i would conditionally append these tools only if one of the systems has a resource capability. I'd also update the tooltip to include which systems have resources